Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -166,4 +171,135 @@ lemma restrict_circuit_iff (hR : R ⊆ M.E := by aesop_mat) :
+    M.fundCircuit e I ⊆ insert e I :=
+  insert_subset_insert inter_subset_left
+
+lemma fundCircuit_subset_ground (he : e ∈ M.E) (hI : I ⊆ M.E := by aesop_mat) :
+    M.fundCircuit e I ⊆ M.E :=
+  (M.fundCircuit_subset_insert e I).trans (insert_subset he hI)
+
+lemma mem_fundCircuit (M : Matroid α) (e : α) (I : Set α) : e ∈ fundCircuit M e I :=
+  mem_insert _ _
+
+/-- The fundamental circuit of `e` and `X` has the junk value `{e}` if `e ∈ X` -/
+lemma fundCircuit_eq_of_mem (heX : e ∈ X) (heE : e ∈ M.E := by aesop_mat) :
+    M.fundCircuit e X = {e} := by
+  suffices h : ∀ a ∈ X, (∀ I ⊆ X, e ∈ M.closure I → a ∈ I) → a = e by
+    simpa [subset_antisymm_iff, fundCircuit]
+  exact fun f hfX h ↦ h {e} (by simpa) (mem_closure_of_mem' _ rfl)
+
+/-- A version of `Matroid.fundCircuit_eq_of_mem` that applies when `X ⊆ M.E` instead of `e ∈ X`.-/
+lemma fundCircuit_eq_of_mem' (heX : e ∈ X) (hX : X ⊆ M.E := by aesop_mat) :
+    M.fundCircuit e X = {e} := by
+  rwa [fundCircuit_eq_of_mem]
+
+lemma Indep.fundCircuit_circuit (hI : M.Indep I) (hecl : e ∈ M.closure I) (heI : e ∉ I) :
+    M.Circuit (M.fundCircuit e I) := by
+  apply (hI.inter_right _).insert_circuit_of_forall (by simp [heI])
+  · rw [(hI.subset _).closure_inter_eq_inter_closure, mem_inter_iff, and_iff_right hecl,
+      hI.closure_sInter_eq_biInter_closure_of_forall_subset _
+      (by simp (config := {contextual := true}))]
+    · simp
+    · exact ⟨I, rfl.subset, hecl⟩
+    exact union_subset rfl.subset (sInter_subset_of_mem ⟨rfl.subset, hecl⟩)
+  simp only [mem_inter_iff, mem_sInter, mem_setOf_eq, and_imp]
+  exact fun f hfI hf hecl ↦ (hf _ (diff_subset.trans inter_subset_left) hecl).2 rfl
+
+lemma Indep.mem_fundCircuit_iff (hI : M.Indep I) (hecl : e ∈ M.closure I) (heI : e ∉ I) :
+    x ∈ M.fundCircuit e I ↔ M.Indep (insert e I \ {x}) := by
+  obtain (rfl | hne) := eq_or_ne x e
+  · simp [hI.diff, mem_fundCircuit]
+  suffices (∀ t ⊆ I, e ∈ M.closure t → x ∈ t) ↔ e ∉ M.closure (I \ {x}) by
+    simpa [fundCircuit_eq_sInter hecl, hne, ← insert_diff_singleton_comm hne.symm,
+      (hI.diff _).insert_indep_iff, mem_ground_of_mem_closure hecl, heI]
+  refine ⟨fun h hecl ↦ (h _ diff_subset hecl).2 rfl, fun h J hJ heJ ↦ by_contra fun hxJ ↦ h ?_⟩
+  exact M.closure_subset_closure (subset_diff_singleton hJ hxJ) heJ
+
+lemma Base.fundCircuit_circuit {B : Set α} (hB : M.Base B) (hxE : x ∈ M.E) (hxB : x ∉ B) :
+    M.Circuit (M.fundCircuit x B) :=
+  hB.indep.fundCircuit_circuit (by rwa [hB.closure_eq]) hxB
+
+/-- For `I` independent, `M.fundCircuit e I` is the only circuit contained in `{e} ∪ I`. -/
+lemma Circuit.eq_fundCircuit_of_subset (hC : M.Circuit C) (hI : M.Indep I) (hCss : C ⊆ insert e I) :
+    C = M.fundCircuit e I := by
+  obtain hCI | ⟨heC, hCeI⟩ := subset_insert_iff.1 hCss
+  · exact (hC.not_indep (hI.subset hCI)).elim
+  suffices hss : M.fundCircuit e I ⊆ C by
+    refine Eq.symm <| (hI.fundCircuit_circuit ?_ (fun heI ↦ ?_)).eq_of_subset_circuit hC hss
```

Comment:
```suggestion
    refine ((hI.fundCircuit_circuit ?_ (fun heI ↦ ?_)).eq_of_subset_circuit hC hss).symm
```

Reply:
In fact, might it be sensible to have a version of `eq_of_subset_circuit` the other way round; analogous to how both `Finset.eq_of_subset_of_card_le` and `Finset.eq_of_superset_of_card_ge` exist? I'd be happy either way, but having this alternate version looks like it could be useful to you...